test/distributed/pipelining/test_schedule.py

```diff
@@ -64,553 +44,7 @@ def _prepare_backward_infra(self, n_microbatches):
-        ]
-
-        class CustomState:
-            def __init__(self):
-                self.i = 0
-
-            def dw_builder(self):
-                """This simulates a function attached to a model with a custom backward.
-                Each call to builder gives a new dw_runner that has some updated state to compute the latest dw.
-                """
-
-                def dw_runner():
-                    # This inner function would be called by PipelineStage during `backward_weight_one_chunk`
-                    print(f"dw called {self.i}th time")
-                    self.i += 1
-
-                return dw_runner
-
-        cs = CustomState()
-
-        # Create a pipeline stage to wrap that submodule
-        chunks = 2
-        input_args = x.chunk(chunks)[0]
-        stages = [
-            PipelineStage(
-                stage_module,
-                stage_idx,
-                n_stages,
-                self.device,
-                input_args=input_args,
-                dw_builder=cs.dw_builder,
-            )
-            for stage_module, stage_idx in zip(stage_modules, stage_indices)
-        ]
-
-        # Attach to a schedule
-        schedule = ScheduleClass(stages, chunks, loss_fn=loss_fn)
-
-        # Run
-        for _ in range(2):
-            # Zero gradients
-            for stage_module in stage_modules:
-                stage_module.zero_grad()
-            if self.rank == 0:
-                schedule.step(x)
-            elif self.rank == self.world_size - 1:
-                losses = []
-                out = schedule.step(target=target, losses=losses)
-            else:
-                schedule.step()
-
-        dist.barrier()
-
-        # Last rank checks result
-        if self.rank == self.world_size - 1:
-            # Check output
-            torch.testing.assert_close(out, ref_out)
-            # Check loss
-            # Since the reduction used in the loss function above is "sum", we use
-            # "sum" here to reduce microbatch losses into a single value too.
-            pipe_loss = sum(losses)
-            torch.testing.assert_close(pipe_loss, ref_loss)
-
-        # Every rank checks gradients
-        for stage_module, submod_name in zip(stage_modules, submod_names):
-            # Get corresponding submodule from reference model
-            ref_submod = ref_mod.get_submodule(submod_name)
-            # Check gradients per parameter
-            for name, p in stage_module.named_parameters():
-                ref_p = ref_submod.get_parameter(name)
-                try:
-                    torch.testing.assert_close(p.grad, ref_p.grad, rtol=1e-5, atol=4e-5)
-                except AssertionError:
-                    print(f"Gradient test failed for {name}: {p.grad} vs {ref_p.grad}")
-                    raise
-
-
-instantiate_parametrized_tests(ScheduleTest)
-
-
-class TestSchedulePlan(unittest.TestCase):
+class TestSchedulePlan(TestCase):
```

Comment:
One alternative is to move this test to its own file since this is the only single process test in this file.

Reply:
well i added more tests to this file in the later PRs.